### PR TITLE
G-API: Addressed various documentation issues

### PR DESCRIPTION
**Pull request by dmatveev**
- Fixed various typos and missing references;
- Added brief documentaion on G_TYPED_KERNEL and G_COMPOUND_KERNEL macros;
- Briefly described GComputationT<>;
- Briefly described G-API data objects (in a group section);
- Better document rendering API;
- Better organized documentation on existing graph compilation arguments.

```
force_builders=Custom
buildworker:Custom=linux-1
build_image:Custom=ubuntu-gapi-freetype:16.04

build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f
```

### CODE REVIEW

**Comment by dmatveev** [comment]
@TolyaTalamanov @AsyaPronina can you please have a look on this? There's a link to generated HTML documentation on the "Docs" builder.

File: modules/gapi/include/opencv2/gapi/render/render.hpp [hunk 1]
```diff
@@ -28,17 +78,21 @@ namespace draw
...
 
 /**
- * A structure allows using freetype library for text rendering
+ * @brief This structure specifies which FreeType font to use by FText primitives.
  */
 struct freetype_font
 {
     /*@{*/
-    std::string path; //!< The path to font file (.ttf)
+    std::string path; //!< The path to the font file (.ttf)
```
Comment (TolyaTalamanov) [inline modules/gapi/include/opencv2/gapi/render/render.hpp]: Could you remove `(.ttf)` It's my mistake. Render works with all font formats. Not only `TrueType`

**Comment by TolyaTalamanov** [comment]
LGTM 👍

File: modules/gapi/include/opencv2/gapi/gtyped.hpp [hunk 2]
```diff
@@ -39,6 +39,46 @@ namespace detail
     auto make_default()->decltype(T{}) {return {};}
 }; // detail
 
+/**
+ * @brief This class is a typed wrapper over a regular GComputation.
+ *
+ * `std::function<>`-like template parameter specifies the graph
+ *  signature so methods so the object's constructor, methods like
+ *  `apply()` and the derived `GCompiledT::operator()` aslo become
```
Comment (AsyaPronina) [inline modules/gapi/include/opencv2/gapi/gtyped.hpp]: **also** become typed

File: modules/gapi/include/opencv2/gapi/gtyped.hpp [hunk 3]
```diff
@@ -39,6 +39,46 @@ namespace detail
...
+ *  typed.
+ *
+ *  There is no need to use cv::gin() or cv::gout() modifiers with
+ *  objects of this class.  Instead, all input arguments are followed
```
Comment (AsyaPronina) [inline modules/gapi/include/opencv2/gapi/gtyped.hpp]: It sounds like a there is no need of using cv::gin(GComputationT())...

File: modules/gapi/include/opencv2/gapi/render/render.hpp [hunk 4]
```diff
@@ -18,6 +18,56 @@
...
 #include <opencv2/gapi/own/scalar.hpp>
 
+
+/** \defgroup gapi_draw G-API Drawing and composition functionality
+ *  @{
+ *
+ *  @brief Functions for in-graph drawing.
+ *
+ *  @note This is a Work in Progress functionality and APIs may
+ *  change in the future releases.
```
Comment (AsyaPronina) [inline modules/gapi/include/opencv2/gapi/render/render.hpp]: and APIs may be changed in the future releases

File: modules/gapi/samples/draw_example.cpp [hunk 5]
```diff
@@ -0,0 +1,57 @@
+#include <opencv2/imgproc.hpp>
+#include <opencv2/videoio.hpp>
```
Comment (AsyaPronina) [inline modules/gapi/samples/draw_example.cpp]: are these needed?

File: modules/gapi/include/opencv2/gapi.hpp [hunk 6]
```diff
@@ -13,11 +13,11 @@
 /** \defgroup gapi G-API framework
 @{
     @defgroup gapi_main_classes G-API Main Classes
-    @defgroup gapi_data_objects G-API Data Objects
+    @defgroup gapi_data_objects G-API Data Types
```
[the remark this answers is not in the document]
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi.hpp]: Not necessary, it is an internal ID

File: modules/gapi/include/opencv2/gapi/core.hpp [hunk 7]
```diff
@@ -18,12 +18,12 @@
...
 #include <opencv2/gapi/gkernel.hpp>
 
-/** \defgroup gapi_core G-API core (basic) functionality
+/** \defgroup gapi_core G-API Core functionality
 @{
     @defgroup gapi_math Graph API: Math operations
     @defgroup gapi_pixelwise Graph API: Pixelwise operations
     @defgroup gapi_matrixop Graph API: Operations on matrices
-    @defgroup gapi_transform Graph API: Geometric, depth and LUT-like image transformations
+    @defgroup gapi_transform Graph API: Image and channel composition functions
```
[the remark this answers is not in the document]
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/core.hpp]: Image composition and channel composition functions

File: modules/gapi/include/opencv2/gapi/fluid/gfluidkernel.hpp [hunk 8]
```diff
@@ -97,23 +97,53 @@ class GAPI_EXPORTS GFluidKernel
...
 
+/**
+ * @brief This structure allows to customize the way how Fluid executes
+ * parallel regions.
+ *
+ * For example, user can utilize his own threading runtime via this parameter.
+ * The `parallel_for` member functor is called by the Fluid runtime with the
+ * following arguments:
+ *
+ * @param size Size of the parallel range to process in this thread
```
[the remark this answers is not in the document]
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/fluid/gfluidkernel.hpp]: Across all threads -- thanks, will fix that

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 9]
```diff
@@ -239,12 +239,31 @@ class GKernelType<K, std::function<R(Args...)> >
...
 //! @endcond
+
+/**
+ * Declares a new G-API Operation. See [Kernel API](@ref gapi_kernel_api)
+ * for more details.
+ *
+ * @param Class type name for this operation.
+ * @param API an `std::function<>`-like signature for the operation;
+ *    return type is a single value.
+ * @param Id string identified for the operation. Must be unique.
```
[the remark this answers is not in the document]
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp]: Right

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 10]
```diff
@@ -239,12 +239,31 @@ class GKernelType<K, std::function<R(Args...)> >
...
 // {body} is to be defined by user
 
+/**
+ * Declares a new G-API Operation. See [Kernel API](@ref gapi_kernel_api)
+ * for more details.
+ *
+ * @param Class type name for this operation.
+ * @param API an `std::function<>`-like signature for the operation;
+ *    return type is a tuple of multiple values.
+ * @param Id string identified for the operation. Must be unique.
```
[the remark this answers is not in the document]
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp]: Fixed

File: modules/gapi/include/opencv2/gapi/gmat.hpp [hunk 11]
```diff
@@ -29,10 +29,25 @@ struct GOrigin;
 /** \addtogroup gapi_data_objects
  * @{
  *
- * @brief Data-representing objects which can be used to build G-API
+ * @brief G-API data objects are used to build G-API
```
[the remark this answers is not in the document]
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/gmat.hpp]: Dropped `are`, thanks

File: modules/gapi/include/opencv2/gapi/gtyped.hpp (hunk 2 above)
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/gtyped.hpp]: thanks

File: modules/gapi/include/opencv2/gapi/gtyped.hpp (hunk 3 above)
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/gtyped.hpp]: Not sure if it really does

File: modules/gapi/include/opencv2/gapi/render/render.hpp (hunk 4 above)
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/render/render.hpp]: "may change" also works.

Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/render/render.hpp]: "may change" also works.

File: modules/gapi/samples/draw_example.cpp (hunk 5 above)
Comment (dmatveev) [reply modules/gapi/samples/draw_example.cpp]: imgproc is required (put a comment), videoio is not (dropped).

BTW @TolyaTalamanov "render.hpp" is hardly usable without including "imgproc.hpp" manually -- why not include "imgproc.hpp" in "render.hpp" by default?

Comment (dmatveev) [reply modules/gapi/samples/draw_example.cpp]: @TolyaTalamanov can you please fix this after this thing is merged?

Comment (TolyaTalamanov) [reply modules/gapi/samples/draw_example.cpp]: Yes sure

Comment (dmatveev) [reply modules/gapi/samples/draw_example.cpp]: Great, thanks!

File: modules/gapi/include/opencv2/gapi/render/render.hpp (hunk 1 above)
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/render/render.hpp]: Can you address it on your own in the following `#include` review?

**Comment by AsyaPronina** [comment]
LGTM 👍

**Comment by dmatveev** [comment]
Thanks!!

**Pull request merged**